### PR TITLE
ref(groupStore): Remove GroupActions and improve typing

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: tests/js/spec/stores/groupStore.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import GroupStore from 'sentry/stores/groupStore';
+import {Group, GroupStats, TimeseriesValue} from 'sentry/types';
+
+const g = (id: string, params?: Partial<Group>) => TestStubs.Group({id, ...params});
```
Comment (evanpurkhiser) [inline tests/js/spec/stores/groupStore.spec.tsx:4]: So many parts of the tests were just straight wrong.

File: static/app/stores/groupStore.tsx [hunk 2]
```diff
@@ -384,45 +383,42 @@ const storeConfig: GroupStoreDefinition = {
...
+    this.trigger(new Set(ids));
   },
 
   onUpdateSuccess(changeId, itemIds, response) {
-    itemIds = this._itemIdsOrAll(itemIds);
+    const ids = this.itemIdsOrAll(itemIds);
 
     this.items.forEach((item, idx) => {
-      if (itemIds.indexOf(item.id) !== -1) {
+      if (ids.includes(item.id)) {
```
Comment (scttcper) [inline static/app/stores/groupStore.tsx:421]: yes 2022 here we come

File: static/app/stores/groupStore.tsx [hunk 3]
```diff
@@ -431,28 +427,28 @@ const storeConfig: GroupStoreDefinition = {
...
-      return map;
-    }, {});
+    const groupStatsMap = response.reduce<Record<string, GroupStats>>(
+      (map, stats) => ({...map, [stats.id]: stats}),
+      {}
+    );
 
     this.items.forEach((item, idx) => {
-      if (itemIds.includes(item.id)) {
+      if (itemIds?.includes(item.id)) {
```
Comment (scttcper) [inline static/app/stores/groupStore.tsx:441]: weird check for null here

File: static/app/plugins/components/issueActions.tsx [hunk 4]
```diff
@@ -301,7 +301,11 @@ class IssueActions extends PluginComponentBase<Props, State> {
   }
 
   onSuccess(data) {
-    GroupActions.updateSuccess(null, [this.getGroup().id], {stale: true});
+    // TODO(ts): This needs a better approach. We splice in this attribute to trigger
+    // a refetch in GroupDetails
+    type StaleGroup = Group & {stale?: boolean};
```
Comment (scttcper) [inline static/app/plugins/components/issueActions.tsx:306]: oh god

**Review by scttcper — APPROVED** [review]

File: static/app/stores/groupStore.tsx (hunk 3 above)
Comment (evanpurkhiser) [reply static/app/stores/groupStore.tsx:441]: nah, itemIds CAN be `undefined` here.

**Pull request merged**